components/dashboard/src/settings/Plans.tsx

```diff
@@ -4,13 +4,461 @@
+                    ? <progress value={currentPlan.hoursPerMonth - accountStatement.remainingHours} max={currentPlan.hoursPerMonth} />
+                    : <progress value="0" max="100" />}
+                <p className="text-sm">
+                    <a className={`text-blue-light hover:underline" ${isChargebeeCustomer ? '' : 'invisible'}`} href="javascript:void(0)" onClick={() => { if (chargebeeClient) chargebeeClient.openPortal(); }}>Billing</a>
+                    {!!accountStatement && Plans.isFreePlan(currentPlan.chargebeeId) && <span className="pl-6">{currency === 'EUR'
+                        ? <>€ / <a className="text-blue-light hover:underline" href="javascript:void(0)" onClick={() => setCurrency('USD')}>$</a></>
+                        : <><a className="text-blue-light hover:underline" href="javascript:void(0)" onClick={() => setCurrency('EUR')}>€</a> / $</>}
+                    </span>}
+                </p>
+            </div>
+            <div className="mt-4 flex space-x-2">{planCards}</div>
+            <div className="flex rounded-md w-2/3 bg-gray-200 p-4 mt-14 mx-auto">
+                <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
+                <span>If you are interested in purchasing a plan for a team, purchase a Team plan with one centralized billing. <a className="underline" href="https://www.gitpod.io/docs/teams/" target="_blank">Learn more</a></span>
+            </div>
+            {!!confirmUpgradeToPlan && <Modal visible={true} onClose={() => setConfirmUpgradeToPlan(undefined)}>
+                <h3>Upgrade to {confirmUpgradeToPlan.name}</h3>
+                <div className="border-t border-b border-gray-200 mt-2 -mx-6 px-6 py-2">
+                    <p className="mt-1 mb-4 text-base">You are about to upgrade to {confirmUpgradeToPlan.name}.</p>
+                    {!Plans.isFreePlan(currentPlan.chargebeeId) && <div className="flex rounded-md bg-gray-200 p-4 mb-4">
+                        <img className="w-4 h-4 m-1 ml-2 mr-4" src={info} />
+                        <span>For this billing cycle you will be charged only the total difference ({(confirmUpgradeToPlan.currency === 'EUR' ? '€' : '$') + (confirmUpgradeToPlan.pricePerMonth - applyCoupons(currentPlan, appliedCoupons).pricePerMonth)}). The new total will be effective from the next billing cycle.</span>
+                    </div>}
+                    <div className="flex rounded-md bg-gitpod-kumquat-light p-4 mb-4">
+                        <img className="w-4 h-4 m-1 ml-2 mr-4" src={exclamation} />
+                        <span className="text-red-700">Total: {(confirmUpgradeToPlan.currency === 'EUR' ? '€' : '$') + confirmUpgradeToPlan.pricePerMonth} per month</span>
+                    </div>
+                </div>
+                <div className="flex justify-end mt-6">
+                    <button onClick={doUpgrade}>Upgrade</button>
```

Comment:
**question**: What do you think of introducing a secondary button variant for the confirm (green color) action? Using the following attributes could could suffice. Feel free to add this inline here.

> `text-green-600 bg-green-100 hover:text-green-600 hover:bg-green-300`

Reply:
Interesting, thanks! Do you think this should be added to our current centralized button variants?

https://github.com/gitpod-io/gitpod/blob/ec62ded26ad4df5a90ab1a49b8fd3d3634ead74d/components/dashboard/src/index.css#L36-L51

Reply:
Ideally yes, but this will become more clear on the way as we split components down to smaller pieces. I'd say both approaches are completely fine here! 🔥